packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -40,6 +40,30 @@ import 'theme_data.dart';
 ///
 /// The [clipBehavior] argument must not be null.
 ///
+/// {@tool sample}
+///
+/// This example shows to use [FlatButton] to create a simple button
+/// in your flutter code.
+///
+/// ```dart
+/// FlatButton(
+///   color: Colors.blue,
+///   textColor: Colors.white,
+///   disabledColor: Colors.grey,
+///   disabledTextColor: Colors.black,
+///   padding: EdgeInsets.all(8.0),
+///   splashColor: Colors.blueAccent,
+///   onPressed: () {
+///     /*...*/
+///   },
+///   child: Text(
+///     "Flat Button",
+///     style: TextStyle(fontSize: 20.0),
+///   ),
+/// )
+/// ```
+/// {@end-tool}
+///
```

Comment:
This is great. I think we can make it even better! I think it would be better with two examples, one showing just a simple button with no special colours or styles, and the second showing the version with all the changes you have here.

The reason to have both is that we would want to encourage most people to use the simple version with no styles, and to style their buttons directly from the `Theme` instead.

Reply:
Sure sir! I'll do the necessary changes to it. :-)

Reply:
I did the changes sir. Can you please review this?